### PR TITLE
Support Google Private IPs

### PR DESCRIPTION
**Pull request by Horneth**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,17 +1,21 @@
...
-types](https://github.com/docker/distribution/blob/05b0ab0/docs/spec/manifest-v2-1.md).
-
-* Added a `/batch` submit endpoint that accepts a single wdl with
-multiple input files.
-
-* The `/query` endpoint now supports querying by `id`, and submitting
-parameters as a HTTP POST.
+## 0.21
 
+* Add support for Google Private IPs through `noAddress` runtime attribute. If set to true, the VM will NOT be provided with a public IP address.
```
Comment (Horneth) [inline CHANGELOG.md]: I was not sure where to document this so I put it there.
Also not sure if we want to keep the 0.20 changelog items ?

Comment (geoffjentry) [inline CHANGELOG.md]: I could see an argument either way. They should at least still exist, although they could be put under 0.21

File: supportedBackends/jes/src/main/scala/cromwell/backend/impl/jes/JesPipelineInfo.scala [hunk 2]
```diff
@@ -5,12 +5,12 @@ import wdl4s.parser.MemoryUnit
 
 import scala.collection.JavaConverters._
 
-sealed trait JesRuntimeInfo {
+sealed trait JesPipelineInfo {
```
Comment (geoffjentry) [inline supportedBackends/jes/src/main/scala/cromwell/backend/impl/jes/JesPipelineInfo.scala]: I like the change but curious what prompted it

File: supportedBackends/jes/src/main/scala/cromwell/backend/impl/jes/Run.scala [hunk 3]
```diff
@@ -28,6 +28,7 @@ object Run {
   private val PollingBackoffFactor = 1.1
   private val AcceptableEvents = Set("start", "pulling-image", "localizing-files", "running-docker", "delocalizing-files", "ok", "fail", "start-shutdown", "preempted")
 
+  val noAddressFieldName = "noAddress"
```
Comment (geoffjentry) [inline supportedBackends/jes/src/main/scala/cromwell/backend/impl/jes/Run.scala]: NoAddressFieldName

File: supportedBackends/jes/src/main/scala/cromwell/backend/impl/jes/Run.scala [hunk 4]
```diff
@@ -44,19 +45,20 @@ object Run {
...
-             .setResources(runtimeInfo.resources)
+             .setDocker(pipelineInfo.docker)
+             .setResources(pipelineInfo.resources)
              .setName(workflow.workflowNamespace.workflow.unqualifiedName)
              .setInputParameters(jesParameters.collect({ case i: JesInput => i.toGooglePipelineParameter }).toVector.asJava)
              .setOutputParameters(jesParameters.collect({ case i: JesFileOutput => i.toGooglePipelineParameter }).toVector.asJava)
 
     def runPipeline: String = {
-      val rpargs = new RunPipelineArgs().setProjectId(projectId).setServiceAccount(JesServiceAccount)
+      val runtimeResources = new PipelineResources().set(noAddressFieldName, runtimeAttributes.noAddress)
```
Comment (geoffjentry) [inline supportedBackends/jes/src/main/scala/cromwell/backend/impl/jes/Run.scala]: couldn't you take the noAddress out of the pipelineInfo? 

And iff you can, then the string would only be in one place so no need to pull it into a constant (NoAddressFieldName)

**Comment by geoffjentry** [comment]
👍

File: supportedBackends/jes/src/main/scala/cromwell/backend/impl/jes/JesPipelineInfo.scala (hunk 2 above)
Comment (Horneth) [inline supportedBackends/jes/src/main/scala/cromwell/backend/impl/jes/JesPipelineInfo.scala]: Google refers to the pipeline as "create time" and the run itself as "runtime", and what this is is the resources for the pipeline, which was confusing.

File: supportedBackends/jes/src/main/scala/cromwell/backend/impl/jes/Run.scala (hunk 4 above)
Comment (Horneth) [inline supportedBackends/jes/src/main/scala/cromwell/backend/impl/jes/Run.scala]: It needs to be in both: `In order to use this, must be true for both create time and run time. Cannot be true at run time if false at create time.`

File: supportedBackends/jes/src/main/scala/cromwell/backend/impl/jes/JesRuntimeAttributes.scala [hunk 5]
```diff
@@ -121,6 +128,16 @@ object JesRuntimeAttributes {
     }
   }
 
+  private def validateNoAddress(noAddress: WdlValue): ErrorOr[Boolean] = {
+    val noAddressValidation = validateBoolean(noAddress)
+    if (noAddressValidation.isFailure) {
```
Comment (mcovarr) [inline supportedBackends/jes/src/main/scala/cromwell/backend/impl/jes/JesRuntimeAttributes.scala]: can this be a `leftMap`?

**Comment by mcovarr** [comment]
:+1:

[Approved with PullApprove](https://pullapprove.com/broadinstitute/cromwell/pull-request/1387/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=broadinstitute/cromwell)

File: supportedBackends/jes/src/main/scala/cromwell/backend/impl/jes/JesRuntimeAttributes.scala (hunk 5 above)
Comment (Horneth) [inline supportedBackends/jes/src/main/scala/cromwell/backend/impl/jes/JesRuntimeAttributes.scala]: sure, TBH it's a copy / paste from one of the other `validate***` methods.

**Pull request merged**